fuel/app/classes/materia/widget/instance.php

```diff
@@ -550,6 +580,63 @@ public function lti_associations()
 			->get();
 	}
 
+	public function get_all_extra_attempts()
+	{
+		$semester = Semester::get_current_semester();
+
+		$result = \DB::select('id', 'user_id', 'context_id','extra_attempts')
+			->from('user_extra_attempts')
+			->where('inst_id', $this->id)
+			->where('semester', $semester)
+			->execute()
+			->as_array();
+
+		return $result;
+	}
+
+	public function set_extra_attempts($user_id, $extra_attempts, $context_id, $id=null)
```

Comment:
Consider a more specific id OR change to the unique key combo of user_id + inst_id 

```suggestion
	public function set_extra_attempts(int $user_id, int $extra_attempts, string $context_id, $id=null)
```